raster/src/main/scala/geotrellis/raster/ArrayTile.scala

```diff
@@ -277,3 +221,67 @@ trait ArrayTile extends Tile with Serializable {
 
   def toBytes: Array[Byte]
 }
+
+object ArrayTile {
+  def alloc(t: CellType, cols: Int, rows: Int): MutableArrayTile =
+    t match {
+      case _: BitCells => BitArrayTile.ofDim(cols, rows)
+      case ct: ByteCells => ByteArrayTile.ofDim(cols, rows, ct)
+      case ct: UByteCells => UByteArrayTile.ofDim(cols, rows, ct)
+      case ct: ShortCells => ShortArrayTile.ofDim(cols, rows, ct)
+      case ct: UShortCells => UShortArrayTile.ofDim(cols, rows, ct)
+      case ct: IntCells => IntArrayTile.ofDim(cols, rows, ct)
+      case ct: FloatCells => FloatArrayTile.ofDim(cols, rows, ct)
+      case ct: DoubleCells => DoubleArrayTile.ofDim(cols, rows, ct)
+    }
+
+  def empty(t: CellType, cols: Int, rows: Int): MutableArrayTile =
+    t match {
+      case _: BitCells => BitArrayTile.empty(cols, rows)
+      case ct: ByteCells => ByteArrayTile.empty(cols, rows, ct)
+      case ct: UByteCells => UByteArrayTile.empty(cols, rows, ct)
+      case ct: ShortCells => ShortArrayTile.empty(cols, rows, ct)
+      case ct: UShortCells => UShortArrayTile.empty(cols, rows, ct)
+      case ct: IntCells => IntArrayTile.empty(cols, rows, ct)
+      case ct: FloatCells => FloatArrayTile.empty(cols, rows, ct)
+      case ct: DoubleCells => DoubleArrayTile.empty(cols, rows, ct)
+    }
+
+  def fromBytes(bytes: Array[Byte], t: CellType, cols: Int, rows: Int): MutableArrayTile =
+    t match {
+      case _: BitCells => BitArrayTile.fromBytes(bytes, cols, rows)
+      case ct: ByteCells => ByteArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: UByteCells => UByteArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: ShortCells => ShortArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: UShortCells => UShortArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: IntCells => IntArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: FloatCells => FloatArrayTile.fromBytes(bytes, cols, rows, ct)
+      case ct: DoubleCells => DoubleArrayTile.fromBytes(bytes, cols, rows, ct)
+    }
+
+  def fromBytes(bytes: Array[Byte], t: CellType, cols: Int, rows: Int, replaceNoData: Double): MutableArrayTile =
```

Comment:
@lossyrob:  + for comment

This isn't actually used anywhere. It seems like since we support user defined nodata values explicitly now there is no use case for this. I started thinking that the name is confusing for this overload but do we even need to provide this now?